pkg/client/cli/cliutil/connector.go

```diff
@@ -84,9 +89,8 @@ func withConnector(ctx context.Context, maybeStart bool, withNotify bool, fn fun
 	ctx = context.WithValue(ctx, connectorConnCtxKey{}, conn)
 	connectorClient := connector.NewConnectorClient(conn)
 	if !started {
-		daemonBinary := client.GetConfig(ctx).Daemons.UserDaemonBinary
 		// Ensure that the already running daemon has the correct version
-		if err := versionCheck(ctx, "User", daemonBinary, connectorClient); err != nil {
+		if err := versionCheck(ctx, "User", client.GetConfig(ctx).Daemons.UserDaemonBinary, connectorClient); err != nil {
```

Comment:
Shouldn't this use the `connectorDaemon` initialized around line 59? The value from the config isn't guaranteed.

Reply:
Yeah which means we won't do this check: https://github.com/telepresenceio/telepresence/blob/release/v2/pkg/client/cli/cliutil/version_check.go#L29-L32 which is what I thought you wanted when the config value wasn't set?

Reply:
Oh, I didn't realize that `versionCheck` compared the binaries too. Then this makes sense.